### ISSUE TITLE
Index search not working for catalogs other than 'portal_catalog' 

### ISSUE THREAD

**Issue by ramonski** [senaite/senaite.jsonapi#23]
## Steps to reproduce

Query an index of a catalog other than portal_catalog, e.g.:

@@API/senaite/v1/search??catalog=bika_analysis_catalog&Title=Weight*

## Current behavior

If the index searched for is does exist in portal_catalog (yes, portal_catalog), than it is ignored.

See the Code here: https://github.com/senaite/senaite.jsonapi/blob/master/src/senaite/jsonapi/catalog.py#L75 

## Expected behavior

The index check uses the right catalog and builds the query accordingly

**Comment by nihadness** [comment]
`/API/senaite/v1/search?catalog=bika_analysis_catalog&title=Bacterial%20Identification` 
works but 
`/API/senaite/v1/search?catalog=bika_analysis_catalog&title=Bacterial*` doesn't.
So in your case it is probably because of **T**itle. But it seems wildcards do not work anyways.

**Comment by ramonski** [comment]
Please see the code above. It filters out any index name that is not present in the queried catalog, but unfortunately uses always `portal_catalog`

**Comment by nihadness** [comment]
Dear @ramonski , my changes must solve the problem about searching by indexes which is the real issue. But that thing about wildcards still won't work.

**Comment by nihadness** [comment]
yes as you said, we can only search by `SearchableText` index in our catalogs, since it is the only `ZCTextIndex` for all catalogs...
So how should it work when user wants to search by another index? E.g: `getKeyword='test*'`. Should the system query by `SearchableText` (let's say Keyword field value is in`SearchableText`) and then check if `getKeyword` field matches that `test*` Regular Expression? Because it might happen that 'test' is not a keyword but a title, and `SearchableText` contains that field as well.... Let me know if I am clear : )
